monai/handlers/utils.py

```diff
@@ -39,3 +42,13 @@ def stopping_fn(engine: Engine):
         return -engine.state.output
 
     return stopping_fn
+
+
+def all_gather(tensor):
+    """
+    All gather the data of tensor value in distributed data parallel.
+    """
+    # create placeholder to collect the data from all processes
+    output = [torch.zeros_like(tensor) for _ in range(dist.get_world_size())]
+    dist.all_gather(output, tensor)
+    return torch.cat(output, dim=0)
```

Comment:
will we have the case that some tensors are on gpu, some are not, what would be the device of the returned? should this utility check the property?

Reply:
Hi @wyli ,

Thanks for your review.
I don't think we have that case, and `dist_all_gather()` should return an error for it.
I added a check for PyTorch distributed module status in the latest commit.

Thanks.